### PR TITLE
Xmlchange

### PR DESCRIPTION
**Pull request by zora-wang**
DSOffice.cs misses the build path for XML documentation file.
solution: add" <documentationfile> ..</>" in DSOffice.csproj to solve the problem so that node description in excel.cs file can be shown correctly

### CODE REVIEW

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -117,9 +117,81 @@
   <resheader name="writer">
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="attributeArgMustBeConstant" xml:space="preserve">
+    <value>An attribute argument must be a constant expression</value>
+  </data>
+  <data name="baseIsExpectedToCallBaseConstructor" xml:space="preserve">
+    <value>'base' is expected here to call base constructor.</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: The keyword 'base' is expected here to call the base constructor.

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 2]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="CheckIfTheElementsInListAreSameType" xml:space="preserve">
+    <value>Checks if all the elements in the specified list have the same type.</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if all the elements in the specified list are of the same type.

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 3]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="ChecksIfLengthsAreSameInMultiDimentionalList" xml:space="preserve">
+    <value>Checks if each of rows in multidimentional list has the same number of elements</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if each of rows in multidimensional list has the same number of elements.

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 4]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="ChecksIfListContainsTheElement" xml:space="preserve">
+    <value>Checks if the list contains the given element</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if the specified element is present in the specified list

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 5]
```diff
@@ -117,9 +117,81 @@
...
+    <value>Checks if all elements in the specified list are true</value>
+  </data>
+  <data name="ChecksListWithUniformDepth" xml:space="preserve">
+    <value>Checks if the list has a uniform depth</value>
+  </data>
+  <data name="closeBracketExpected" xml:space="preserve">
+    <value>')' expected - Imcomplete Closure</value>
+  </data>
+  <data name="ContainsKeys" xml:space="preserve">
+    <value>Checks if a list contains a specific key</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Checks if the specified key is present in the specified list

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 6]
```diff
@@ -117,9 +117,81 @@
...
+    <value>Determines whether two object instances are equal</value>
+  </data>
+  <data name="emptyCharacterLiteral" xml:space="preserve">
+    <value>Empty character literal.</value>
+  </data>
+  <data name="failedToImport" xml:space="preserve">
+    <value>Failed to import {0}</value>
+  </data>
+  <data name="functionCallCannotBeAtLeftSide" xml:space="preserve">
+    <value>function call is not allowed at the left hand side of an assignment</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: A function call is not allowed to the left-hand side of an assignment

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 7]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
   <data name="FUNCTION_GROUP_RESOLUTION_FAILURE" xml:space="preserve">
     <value>No function called {0} could be found. Please check the name of the function.</value>
   </data>
+  <data name="getKeys" xml:space="preserve">
+    <value>Gets all keys from an list</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Gets all keys from the specified key-value pair list

File: src/Engine/ProtoCore/Lang/BuiltInMethods.cs [hunk 8]
```diff
@@ -795,7 +830,8 @@ public BuiltInMethods(Core core)
                     {
                         new KeyValuePair<string, ProtoCore.Type>("object", TypeSystem.BuildPrimitiveTypeObject(PrimitiveType.kTypeVar, 0)),
                     },
-                    ID = BuiltInMethods.MethodID.kGetType
+                    ID = BuiltInMethods.MethodID.kGetType,
+                    MethodAttributes = new MethodAttributes(){Description = Resources.gettypes}
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Lang/BuiltInMethods.cs]: Follow the same style everywhere... "gettypes" starts with lower case, whereas "ImportFileByGivenFilePathWithMode" starts with upper case, I'll prefer they start with upper case everywhere.

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 9]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="gettypes" xml:space="preserve">
+    <value>Gets the Type of the current instance</value>
+  </data>
+  <data name="getValues" xml:space="preserve">
+    <value>Gets all values that stored in the list, including in dictionary</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Gets all values stored in the specified key-value pair list and for a simple list it returns all elements in the list

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 10]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="ImportFileByGivenFilePath" xml:space="preserve">
+    <value>Imports from CSV by providing file path</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Imports data from a text file containing comma separated values into two-dimensional list

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 11]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="ImportFileByGivenFilePathWithMode" xml:space="preserve">
+    <value>Imports from CSV by providing file path. The result is transposed when trans is false</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Imports data from a text file containing comma separated values into two-dimensional list and also transpose the output list if specified

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 12]
```diff
@@ -117,9 +117,81 @@
...
+  </data>
+  <data name="importStatementNotSupported" xml:space="preserve">
+    <value>Import statements are not supported in CodeBlock Nodes.</value>
+  </data>
+  <data name="InsertsAnElementIntoList" xml:space="preserve">
+    <value>Inserts an element into a list at the specified index</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Inserts an element into a list at specified index

File: src/Engine/ProtoCore/Properties/Resources.en-US.resx [hunk 13]
```diff
@@ -174,6 +246,12 @@
...
     <value>Exception type '{0}' is not defined.</value>
   </data>
+  <data name="keywordCannotBeUsedAsConstructorName" xml:space="preserve">
+    <value>\"{0}\" is a keyword, can't be used as constructor name</value>
+  </data>
+  <data name="keywordCantBeUsedAsIdentifier" xml:space="preserve">
+    <value>\"{0}\" is a keyword, identifier expected</value>
+  </data>
   <data name="kFailToConverToFunction" xml:space="preserve">
     <value>Converting other things to function pointer is not allowed.</value>
```
Comment (sharadkjaiswal) [inline src/Engine/ProtoCore/Properties/Resources.en-US.resx]: Converting other objects to a function pointer is not allowed.

**Comment by zora-wang** [comment]
@sharadkjaiswal , 
Grammars have been corrected.

**Comment by sharadkjaiswal** [comment]
LGTM

**Pull request merged**